### PR TITLE
NAS-104165 / 12 / Platform specific system.info implementation

### PR DESCRIPTION
**Pull request by sonicaj**

### CODE REVIEW

File: src/middlewared/middlewared/utils/__init__.py [hunk 1]
```diff
@@ -331,6 +331,10 @@ def sw_version_is_stable():
     return version['stable']
 
 
+def cpu_model():
```
Comment (themylogin) [inline src/middlewared/middlewared/utils/__init__.py]: What for?

File: src/middlewared/middlewared/utils/osc/linux/system.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+logger = logging.getLogger(__name__)
+
+__all__ = ['cpu_model']
+
+
+RE_CPU_MODEL = re.compile(r'Model name:\s*(.*)')
+
+
+def cpu_model():
+    cp = subprocess.Popen('lscpu', stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```
Comment (themylogin) [inline src/middlewared/middlewared/utils/osc/linux/system.py]: You can just read from `/proc/cpuinfo`

File: src/middlewared/middlewared/utils/__init__.py [hunk 3]
```diff
@@ -331,6 +331,10 @@ def sw_version_is_stable():
     return version['stable']
 
 
+def get_cpu_model():
```
Comment (themylogin) [inline src/middlewared/middlewared/utils/__init__.py]: I am still not convinced this is necessary

Comment (sonicaj) [reply src/middlewared/middlewared/utils/__init__.py]: I wasn't sure about it myself. Should we use `osc` directly in system plugin ?

Comment (themylogin) [reply src/middlewared/middlewared/utils/__init__.py]: Yes, we import from osc directly

File: src/middlewared/middlewared/plugins/system.py [hunk 4]
```diff
@@ -7,6 +7,7 @@
 from middlewared.service import CallError, ConfigService, no_auth_required, job, private, Service, ValidationErrors
 import middlewared.sqlalchemy as sa
 from middlewared.utils import Popen, run, start_daemon_thread, sw_buildtime, sw_version
+from middlewared.utils import osc
```
Comment (william-gr) [inline src/middlewared/middlewared/plugins/system.py]: Combine with line above please

Comment (sonicaj) [reply src/middlewared/middlewared/plugins/system.py]: oops, sorry

**Pull request merged**